pkg/collector/reconcile/route.go

```diff
@@ -75,11 +74,10 @@ func desiredRoutes(_ context.Context, params Params) []routev1.Route {
 				Path: "/" + p.Name,
 				To: routev1.RouteTargetReference{
 					Kind: "Service",
-					Name: naming.Service(params.Instance),
+					Name: naming.Service(params.Instance.Name),
 				},
 				Port: &routev1.RoutePort{
-					// Valid names must be non-empty and no more than 15 characters long.
-					TargetPort: intstr.FromString(naming.Truncate(p.Name, 15)),
+					TargetPort: intstr.FromString(naming.PortName(p.Name, p.Port)),
```

Comment:
I might be missing something, but it seems the new code doesn't enforce a 15 character limit anymore - I didn't see anything in `naming.PortName` that is truncating this value to 15 characters.

Reply:
correct, the 15 character limit should not be applied to service/route/ingress objects. It is only required for port names in the pod spec. 

I have documented it in the `naming` package.